### ISSUE TITLE
Out parameter on Dictionary<TKey, TValue>.TryGetValue(TKey, out TValue) is flagged by IDE0008 when it shouldn't be

### ISSUE THREAD

**Issue by NetMastr5** [dotnet/roslyn#74372]
**Version Used**: Visual Studio Enterprise 17.10.4

**Steps to Reproduce**:

```c#
using System.Linq;

namespace ClassLibrary1
{
    public class Temp
    {
		public void temp()
		{
			var y = new[]{ new { t = 0 }}.ToList();

			y.ToDictionary(x => x.t, x => x).TryGetValue(0, out var y2);
		}
    }
}
```

**Diagnostic Id**: IDE0008

If this is a report about a bug in an analyzer, please include the diagnostic ID and message if possible (e.g. `"IDE0030: Use coalesce expression"`).

**Expected Behavior**:  out variable should not be flagged, and code change suggestion is incorrect as it tries to change it to object

**Actual Behavior**:  out variable is flagged

![image](https://github.com/user-attachments/assets/de18c196-d711-4559-b37e-b261958b3c0a)
